daemon/api_icons_test.go

```diff
@@ -98,7 +127,34 @@ func (s *iconsSuite) TestAppIconGetNoIcon(c *check.C) {
 	c.Check(rec.Code/100, check.Equals, 4)
 }
 
-func (s *iconsSuite) TestAppIconGetNoApp(c *check.C) {
+func (s *iconsSuite) TestSnapIconGetFallback(c *check.C) {
+	d := s.daemon(c)
+
+	const snapName = "foo"
+	const snapID = "foo-id" // ID is "<name>-id" for apiBaseSuite
+
+	info := s.mkInstalledInState(c, d, snapName, "bar", "v1", snap.R(10), true, "")
```

Comment:
for completeness, can you add a test with an unasserted snap? the snap ID could be set to  `snap.R(-1)`